src/library/implementation_interface.hpp

```diff
@@ -299,12 +345,19 @@ struct SerialImplementation {
         return 0;
     };
     static void close(){};
-    static std::tuple<hid_t, hid_t, hid_t, std::string> prepare_write(
-        const std::string& report_name) {
-        return std::make_tuple(H5Pcreate(H5P_FILE_ACCESS),
-                               H5Pcreate(H5P_DATASET_XFER),
-                               H5Pcreate(H5P_DATASET_XFER),
-                               report_name + FILE_EXTENSION);
+    static void add_communicator(const std::string& /*comm_nam*/){};
```

Comment:
nitpick. We could have some docs for each public interface function, at least explaining what it's supposed to do and to help splitting the code.

Reply:
Totally agree. Will add this to the TODO and will do it in the next PR when I will improve other things also.